[Build 113](https://ci.wildfly.org/viewLog.html?buildId=169412&buildTypeId=Xnio_37_LinuxIbmJdk8) outcome was **FAILURE** using a merge of 02c95d67079e1d14ee8cd51f54cf18fce2cf5113
Summary: Tests failed: 7 (1 new), passed: 945, ignored: 18 Build time: 00:01:28
### Failed tests
```
org.xnio.nio.test.MultiThreadedNioSslTcpConnectionTestCase.twoWayTransfer: java.lang.AssertionError
------- Stderr: -------
06:11:14,999 INFO  (main) [TEST] <AbstractNioSslTcpTest.java:231> Test: twoWayTransfer
06:11:15,008 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:255> client handle readable
06:11:15,016 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,017 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,017 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,017 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,018 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,018 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,018 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,019 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,019 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,019 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,019 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,020 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,020 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,020 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,021 INFO  (XNIO-45 I/O-1) [TEST] <AbstractNioSslTcpTest.java:336> server handle readable
06:11:15,021 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,021 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,021 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,022 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,022 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,022 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,023 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,023 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,023 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,023 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,024 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,024 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,026 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,026 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,027 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,027 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,027 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,028 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,028 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,028 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,029 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,029 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,029 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,030 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,030 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,030 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,030 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,031 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,031 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,031 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,032 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable
06:11:15,032 INFO  (XNIO-45 I/O-2) [TEST] <AbstractNioSslTcpTest.java:273> client handle writable


org.xnio.racecondition.ResumeReadsOnHandlingReadableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)


org.xnio.racecondition.ResumeWritesOnHandlingWritableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)


org.xnio.racecondition.SetReadListenerOnHandlingReadableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)


org.xnio.racecondition.SetReadReadyOnHandlingReadableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)


org.xnio.racecondition.SetWriteListenerOnHandlingWritableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)


org.xnio.racecondition.SetWriteReadyOnHandlingWritableChannelTestCase: java.net.ConnectException: Connection refused (Connection refused)
------- Stdout: -------
byteman jar is /store/repository/org/jboss/byteman/byteman/3.0.3/byteman-3.0.3.jar
Setting org.jboss.byteman.allow.config.update=true
------- Stderr: -------
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:408)
	at sun.instrument.InstrumentationImpl.loadClassAndCallAgentmain(InstrumentationImpl.java:433)
	at com.ibm.tools.attach.javaSE.Attachment.loadAgentLibraryImpl(Native Method)
	at com.ibm.tools.attach.javaSE.Attachment.loadAgentLibrary(Attachment.java:309)
	at com.ibm.tools.attach.javaSE.Attachment.parseLoadAgent(Attachment.java:287)
	at com.ibm.tools.attach.javaSE.Attachment.doCommand(Attachment.java:168)
	at com.ibm.tools.attach.javaSE.Attachment.run(Attachment.java:128)
Caused by: java.lang.UnsupportedOperationException: adding retransformable transformers is not supported in this environment
	at sun.instrument.InstrumentationImpl.addTransformer(InstrumentationImpl.java:100)
	at org.jboss.byteman.agent.Main.premain(Main.java:275)
	at org.jboss.byteman.agent.Main.agentmain(Main.java:305)
	... 11 more


```

--- a/api/src/main/java/org/xnio/AbstractIoFuture.java
+++ b/api/src/main/java/org/xnio/AbstractIoFuture.java
@@ -719,7 +719,7 @@ public abstract class AbstractIoFuture<T> implements IoFuture<T> {
             try {
                 notifier.notify(future, attachment);
             } catch (Throwable t) {
-                futureMsg.notifierFailed(t, notifier);
+                futureMsg.notifierFailed(t, notifier, attachment);
             }
         }
     }

--- a/api/src/main/java/org/xnio/_private/Messages.java
+++ b/api/src/main/java/org/xnio/_private/Messages.java
@@ -310,9 +310,9 @@ public interface Messages extends BasicLogger {
     @Message(id = 1002, value = "Operation was cancelled")
     CancellationException opCancelled();
 
-    @Message(id = 1003, value = "Running IoFuture notifier %s failed")
+    @Message(id = 1003, value = "Running IoFuture notifier %s (with attachment %s) failed")
     @LogMessage(level = WARN)
-    void notifierFailed(@Cause Throwable cause, IoFuture.Notifier<?, ?> notifier);
+    void notifierFailed(@Cause Throwable cause, IoFuture.Notifier<?, ?> notifier, Object attachment);
 
     @Message(id = 1004, value = "Operation timed out")
     TimeoutException opTimedOut();

